### PR TITLE
Add storage-independent media layer with R2 and Gyazo

### PR DESCRIPTION
**Pull request by nomadoor**
## Changes
- Add a storage-independent media layer. `resolveMedia(url, { mode, size })` in `.eleventy.js` is the only place that knows about R2 or Gyazo; Markdown images, `{media}`/`{gyazo}` embeds, `mediaRow`, hero, OGP, related cards, Notes finder cards, and the assistant rail video all use it.
- Serve R2 media from `media.comfyui.nomadoor.net` through logical references: `{media=image}`. `src/_data/media.json` maps logical names to content-hashed keys (`images/<hash>.<ext>`, `videos/<hash>.mp4`).
- Add `{media=image|loop|player}` as the display-mode syntax for any source; `{gyazo=…}` remains a compatible alias. Gyazo stays a supported source.
- Rename storage-specific UI hooks (`gyazo-toggle` → `media-toggle`, `data-gyazo-*` → `data-media-*`); `lightbox.js` now reads only `data-full-src`.
- Add `npm run media:put`: takes a path relative to `COMFY_MEDIA_ORIGINALS`, strips ComfyUI workflow/EXIF/XMP/comment metadata without re-encoding (rejects orientation/Ultra HDR cases), uploads with `Cache-Control: public, max-age=31536000, immutable`, records the entry, and copies the Markdown. Supports `--replace`, `--force`, `--dry-run`.
- Add `check:media` to `npm run check`: validates names, keys, display modes, unregistered `/media/` references, and physical R2 URLs; lists unused entries as advisories. Unregistered `/media/` names also fail production builds.
- Separate test fixtures from production: fixture manifest and page live in `tests/fixtures/media/` and load only with `COMFY_MEDIA_FIXTURES=1`. Playwright uses a dedicated `dev:test` server (port 8091, `.cache/playwright-site`) and fulfills Gyazo/R2 requests from local fixtures.
- CI now runs `npm run check` → build → Playwright.
- Register the first real R2 image (MiniMax H3).

## Why
Gyazo-hosted images became unavailable. This lets affected pages be restored with self-hosted R2 media while keeping Markdown readable and independent of the storage layout, and removes Gyazo-specific logic from templates and client JS.

## Impact
- Existing Gyazo pages render the same, except the renamed UI hooks, added `data-full-src` on videos, and OGP for `wan-2-1-vace` (ja/zh) now uses the Gyazo `max_size` still.
- R2 setup verified on the real bucket: upload, 200 / `image/png` / immutable Cache-Control, Cloudflare cache HIT, served bytes match the key, local rendering and lightbox via R2, and Windows browser display. Bucket Lock (30 days) is enabled.
- See `ops/adr/2026-09-17-media-layer-r2-gyazo.md`.

## Validation
- git diff --check
- npm run check
- npm run build (clean output) + npm run check:build
- npm run test:playwright: 31/31 passed on three consecutive runs; Notes suite 20/20 with `--repeat-each=5`
- Build output compared with `main` for all pages
- Negative checks for `check:media`, uploader edge cases (fake wrangler), and production-build failure on unregistered `/media/` names

## Notes
- Existing advisories remain: 63 Markdown links and 22 icons. `check:media` reports the MiniMax H3 image as unused until an article references it.
- Follow-up PR: rename workflow JSON files to lowercase snake_case with redirects.
- mp4 upload is manual for now (strip metadata with ffmpeg first; see ADR).

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **New Features**
  * Added support for R2-hosted images and videos alongside Gyazo media.
  * Introduced unified media embeds with image, loop, and player display modes.
  * Added video posters, responsive media rendering, and improved lightbox behavior.
  * Added tooling for uploading media and registering logical media references.

* **Bug Fixes**
  * Improved hero, related-content, and social preview handling for image and video media.
  * Added validation for missing, invalid, or incorrectly referenced media.

* **Documentation**
  * Updated authoring guidance and requirements for R2/Gyazo embeds, naming, accessibility, and styling.

### CODE REVIEW

File: .eleventy.js [hunk 1]
```diff
@@ -879,43 +1000,21 @@ export default function (eleventyConfig) {
...
-    const commonFig = `<figure class="article-media" style="--article-media-width:${imgWidth}px; --article-media-height:${imgHeight}px; --article-media-aspect:${imgAspect};"><div class="article-media__frame">`;
-    const closing = `${alt ? `<figcaption>${alt}</figcaption>` : ""}</figure>`;
-    return `${commonFig}<img src="${variants.preview}" data-full-src="${variants.full}" alt="${alt}" loading="lazy" decoding="async" width="${imgWidth}" height="${imgHeight}" /></div>${closing}`;
+    // Display box is fitted to 1200px on the longest side regardless of the media source.
+    const { width, height } = getPreviewDimensions(media, 1200);
+    const figureStyle = `--article-media-width:${width}px; --article-media-height:${height}px; --article-media-aspect:${width} / ${height};`;
+    const caption = alt ? `<figcaption>${alt}</figcaption>` : "";
+    return `<figure class="article-media" style="${figureStyle}"><div class="article-media__frame"><img src="${escapeHTML(media.src)}" data-full-src="${escapeHTML(media.fullSrc)}" alt="${alt}" loading="lazy" decoding="async" width="${width}" height="${height}" /></div>${caption}</figure>`;
```
[the remark this answers is not in the document]
Comment (nomadoor) [reply .eleventy.js:1013]: Not changing this in this PR. `renderGyazoMedia()` on `main` did not emit `srcset` for `{gyazo=image}` either, and this PR intentionally keeps rendered output identical for existing Gyazo embeds (verified by diffing the build against `main`). Adding it would change image selection for ~2,000 embeds. Responsive variants (`srcset` for both R2 and Gyazo) will be handled in the follow-up PR that introduces resized WebP variants.

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -28,11 +28,15 @@ jobs:
...
         run: npm ci
 
-      # 4. ビルド
+      # 4. 整合性チェック（frontmatter / nav / workflow / media など）
+      - name: Run checks
+        run: npm run check
+
+      # 5. ビルド
       - name: Build site
         run: npm run build
```
[the remark this answers is not in the document]
Comment (nomadoor) [reply .github/workflows/ci.yml:37]: Fixed in 9ea5ad0: CI now runs `npm run check:build` right after `npm run build` (check → build → check:build → Playwright).

File: scripts/lib/media-metadata.mjs [hunk 3]
```diff
@@ -0,0 +1,276 @@
...
+  }
+  const chunks = [];
+  let offset = 8;
+  while (offset + 12 <= buffer.length) {
+    const length = buffer.readUInt32BE(offset);
+    const type = buffer.toString("latin1", offset + 4, offset + 8);
+    const end = offset + 12 + length;
+    if (end > buffer.length) throw new MediaMetadataError(`PNG チャンク ${type} が途中で切れています`);
+    const crc = buffer.readUInt32BE(end - 4);
+    if (zlib.crc32(buffer.subarray(offset + 4, end - 4)) !== crc) {
```
[the remark this answers is not in the document]
Comment (nomadoor) [reply scripts/lib/media-metadata.mjs:57]: Fixed in 9ea5ad0: declared `"engines": { "node": ">=20.15.0" }` in `package.json` (and the lockfile). CI uses `node-version: 20`, which resolves to the latest 20.x and satisfies this; the Cloudflare Pages build does not load this module.

**Comment by nomadoor** [comment]
Re the outside-diff finding on `ops/requirements.md` (Social Cards): fixed in 9ea5ad0. The rule now says `og:image` / `twitter:image` use the resolved hero still image (`resolveMedia().poster`; the poster for video heroes) and fall back to `site.ogImage` only when the hero is missing or a video has no poster.
